pageserver/src/tenant/remote_timeline_client/index.rs

```diff
@@ -21,31 +22,41 @@ use utils::lsn::Lsn;
 #[cfg_attr(test, derive(Default))]
 pub struct LayerFileMetadata {
     file_size: Option<u64>,
+    holes: Option<Vec<Hole>>,
```

Comment:
As noted earlier, would suggest going with `Vec<Hole>`.

Reply:
> I wonder if we could separate at least the ctx changes from the rest...
How?
I still think that TLS is better solution.
Or at least ctx should be added to all methods. Adding it on demand IMHO is most awful approach.

Reply:
> As noted earlier, would suggest going with Vec<Hole>.

Sorry, I remembered this your suggestion. But failed to find a reference now. And do not remember arguments. Yes, Option can be eliminated and replaced with empty vector. But why it should be done? To save one byte and one check? It seems to be that Option better express situation when information about holes is not present or is not relevant (for example for image layers). And some minor overhead can not be measurable.

Reply:
First mentions here: https://neondb.slack.com/archives/C0277TKAJCA/p1674829578261489?thread_ts=1674828847.998759&cid=C0277TKAJCA

I did try to ask about it here: https://neondb.slack.com/archives/C033RQ5SPDH/p1674810428051729?thread_ts=1674809569.138119&cid=C033RQ5SPDH based on your answer on the next message I settled with "Option<Vec<_>> is not needed".

This is not a space optimization, because there most likely is no space optimization, or if there is, it's one usize (bringing the size of a Option<Vec<_>> to 4 usizes).

> But why it should be done?

It comes down to what do you want to communicate with this. Do we need to distinguish the "value not set" and "empty vec". Usually one does not, like you answered in slack. 

This would be different if we were able to express `NonEmptyVec<_>` ergonomically but no one usually has time or energy, so we settle with `Option<Vec<_>>` made into `Vec<_>`, because all collections can play the part of "empty" and the reader has less to think about.

In Java circles, I remember a lot of blog posts on the topic "never return null for collection", which is a bit different angle because of the implicit nullability but still the same story.